pilot/cmd/pilot-agent/main.go

```diff
@@ -78,13 +78,14 @@ var (
 	loggingOptions         = log.DefaultOptions()
 	outlierLogPath         string
 
-	instanceIPVar        = env.RegisterStringVar("INSTANCE_IP", "", "")
-	podNameVar           = env.RegisterStringVar("POD_NAME", "", "")
-	podNamespaceVar      = env.RegisterStringVar("POD_NAMESPACE", "", "")
-	kubeAppProberNameVar = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
-	serviceAccountVar    = env.RegisterStringVar("SERVICE_ACCOUNT", "", "Name of service account")
-	clusterIDVar         = env.RegisterStringVar("ISTIO_META_CLUSTER_ID", "", "")
-	callCredentials      = env.RegisterBoolVar("CALL_CREDENTIALS", false, "Use JWT directly instead of MTLS")
+	instanceIPVar          = env.RegisterStringVar("INSTANCE_IP", "", "")
+	podNameVar             = env.RegisterStringVar("POD_NAME", "", "")
+	podNamespaceVar        = env.RegisterStringVar("POD_NAMESPACE", "", "")
+	kubeAppProberNameVar   = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
+	serviceAccountVar      = env.RegisterStringVar("SERVICE_ACCOUNT", "", "Name of service account")
+	clusterIDVar           = env.RegisterStringVar("ISTIO_META_CLUSTER_ID", "", "")
+	callCredentials        = env.RegisterBoolVar("CALL_CREDENTIALS", false, "Use JWT directly instead of MTLS")
+	xdsEnableTokenExchange = env.RegisterBoolVar("XDS_ENABLE_TOKEN_EXCHANGE", false, "Enable token exchange for XDS")
```

Comment:
Do we actually need a flag? For CA, its auto detected: https://github.com/istio/istio/blob/d517e26e3dc38437da762bdc1c57149db55cc07f/pilot/cmd/pilot-agent/main.go#L375-L377

Long term we want a proper API; in the short term should we do auto detection instead of yet another variable?

cc @liminw @costinm

Reply:
For the CA flow, the CAProviderName can be used to determine whether token exchange is enabled or not.
For the XDS flow, I am unaware a flag that can be used to determine whether the token exchange should be enabled or not. Currently, for XDS flow, the token exchange is always disabled. If you have a suggestion on how to do auto detection of token exchange for XDS flow, please let us know.

Reply:
`strings.Contains(o.CAEndpoint, "googleapis.com")`

we can do similar with the xds address? presumable any `googleapis.com` will require token exchange

Reply:
".googleapis.com" for less corner cases; But it's not quite OK to hardcode that - for local testing, this could be localhost or some in-K8s proxy.

Reply:
Thats true, the CA check has `or` that covers this. Ok, we can add the flag as is I guess, but with the expectation that we will have a proper API ASAP.

Also this needs a test under `TestAgent` in pkg/istio-agent

Reply:
I just added a similar comment. I agree we should use XDS address (e.g., containing "googleapis.com") to determine if token exchange is needed.

Reply:
This would destroy our ability to test/develop locally.